### PR TITLE
Adjust cardinality language

### LINKED ISSUE #8: WGLC: Comment regarding cardinality
From Chris Patton:

> "Use of ECH yields an anonymity set of cardinality equal to the number of ECH-enabled server domains supported by a given client-facing server" ... This is only true when certain conditions are meant. Suppose for example that the client-facing server maps one of the backend server domains to a unique IP address not used for any other backend server: this would reduce the anonymity set by 1.

### PR DESCRIPTION
**Pull request by bemasc**
Adjusts this paragraph for clarity, utility, and correctness. 

Fixes #8

### CODE REVIEW

**Review by MikeBishop — APPROVED** [review]

File: draft-ietf-tls-svcb-ech.md [hunk 1]
```diff
@@ -82,7 +82,7 @@ If all HTTPS records for an alt-authority contain "ech" SvcParams, the client MU
...
 
-Use of ECH yields an anonymity set of cardinality equal to the number of ECH-enabled server domains supported by a given client-facing server. Thus, even with an encrypted ClientHello, an attacker who can enumerate the set of ECH-enabled domains supported by a client-facing server can guess the correct SNI with probability at least 1/K, where K is the size of this ECH-enabled server anonymity set. This probability may be increased via traffic analysis or other mechanisms.
+Use of ECH yields an anonymity set whose cardinality is at most the number of ECH-enabled server domains supported by a given client-facing server. Thus, even with an encrypted ClientHello, an attacker who can enumerate the set of ECH-enabled domains supported by a client-facing server can guess the correct SNI with probability at least 1/K, where K is the size of this ECH-enabled server anonymity set. This probability may be increased via traffic analysis or other mechanisms.
```
Comment (enygren) [inline draft-ietf-tls-svcb-ech.md]: I wonder if "at best" or "roughly proportional to" would be better than "at most"?
I'm having trouble understanding exactly what is meant by "at most".
The "traffic analysis and other mechanisms" does help clarify, but it is squishy.

Comment (bemasc) [reply draft-ietf-tls-svcb-ech.md]: OK, I reworked this paragraph to try to make it more intelligible and accurate.  Please review.

File: draft-ietf-tls-svcb-ech.md [hunk 2]
```diff
@@ -82,7 +82,7 @@ If all HTTPS records for an alt-authority contain "ech" SvcParams, the client MU
...
 
-Use of ECH yields an anonymity set of cardinality equal to the number of ECH-enabled server domains supported by a given client-facing server. Thus, even with an encrypted ClientHello, an attacker who can enumerate the set of ECH-enabled domains supported by a client-facing server can guess the correct SNI with probability at least 1/K, where K is the size of this ECH-enabled server anonymity set. This probability may be increased via traffic analysis or other mechanisms.
+In an idealized deployment, ECH protects the SNI with an anonymity set consisting of all the ECH-enabled server domains supported by a given client-facing server. Accordingly, an attacker who can enumerate this set can always guess the encrypted SNI with probability 1/K, where K is the number of domains in the set. In practice, this probability may be increased via traffic analysis, popularity weighting, and other mechanisms.
```
Comment (enygren) [inline draft-ietf-tls-svcb-ech.md:85]:
```suggestion
In an idealized deployment, ECH protects the SNI with an anonymity set consisting of all the ECH-enabled server domains supported by a given client-facing server (i.e. for a given pairing of ClientHelloOuter server_name and IP address). Accordingly, an attacker who can enumerate this set can always guess the encrypted SNI with probability 1/K, where K is the number of domains in the set. In practice, this probability may be increased via traffic analysis, popularity weighting, and other mechanisms.
```

Is this useful to provide more context on what the scope of the anonymity set is?

**Review by enygren — COMMENTED** [review]
(If you think this is more detail than is needed, this otherwise looks fine.)

File: draft-ietf-tls-svcb-ech.md (hunk 2 above)
Comment (bemasc) [reply draft-ietf-tls-svcb-ech.md:85]: Making this fully precise is hard. For example, two ECHConfigs could be distinguishable despite using the same public_name.  I would prefer to lean on the idea that the "idealized deployment" covers things like "only one ECHConfig is in use" and "all servers in the fleet receive an identical mix of inner SNIs".

**Pull request merged**